library/std/src/io/mod.rs

```diff
@@ -917,6 +917,13 @@ pub trait Read {
     /// # }
     /// ```
     ///
+    /// # Notes
+    ///
+    /// Be careful using this trait method with streams that are expected to be continuous. For example, using 
+    /// `read_to_end` with streams like `stdin` will simply lock the application into waiting on the 
+    /// transmission of data to conclude. It is recommended you use this method if you know the stream will be closed
+    /// at the other end. The problem is that EOF or End of File is never reached for streams that never close or are finite.
```

Comment:
I think the wording could be simplified some. Perhaps something like this, with some cleanup:

> # Usage Notes
>
> `read_to_end` attempts to read a source until EOF, but many sources are continuous streams that do not send EOF; in these cases, `read_to_end` will block indefinitely. Standard input is one such stream which may be finite if piped, but is typically continuous without EOF when reading user input from a shell.
>
> Using `.lines()` with a [`BufReader`] can provide a better solution.